tests/warp/warp_test.go

```diff
@@ -122,301 +137,551 @@ var _ = ginkgo.BeforeSuite(func() {
+	require.NoError(err)
+
+	packedInput, err := warp.PackGetVerifiedWarpBlockHash(0)
+	require.NoError(err)
+	tx := predicate.NewPredicateTx(
+		w.receivingSubnetChainID,
+		nonce,
+		&warp.Module.Address,
+		5_000_000,
+		big.NewInt(225*params.GWei),
+		big.NewInt(params.GWei),
+		common.Big0,
+		packedInput,
+		types.AccessList{},
+		warp.ContractAddress,
+		w.blockPayloadSignedMessage.Bytes(),
+	)
+	signedTx, err := types.SignTx(tx, w.receivingSubnetSigner, w.receivingSubnetFundedKey)
+	require.NoError(err)
+	txBytes, err := signedTx.MarshalBinary()
+	require.NoError(err)
+	log.Info("Sending getVerifiedWarpBlockHash transaction", "txHash", signedTx.Hash(), "txBytes", common.Bytes2Hex(txBytes))
+	err = client.SendTransaction(ctx, signedTx)
+	require.NoError(err)
+
+	log.Info("Waiting for new block confirmation")
+	<-newHeads
+	receiptCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	blockHash, _ := w.getBlockHashAndNumberFromTxReceipt(receiptCtx, client, signedTx)
+	log.Info("Fetching relevant warp logs and receipts from new block")
+	logs, err := client.FilterLogs(ctx, interfaces.FilterQuery{
+		BlockHash: &blockHash,
+		Addresses: []common.Address{warp.Module.Address},
 	})
+	require.NoError(err)
+	require.Len(logs, 0)
+	receipt, err := client.TransactionReceipt(ctx, signedTx.Hash())
+	require.NoError(err)
+	require.Equal(receipt.Status, types.ReceiptStatusSuccessful)
+}
 
-	// Verify successful delivery of the Avalanche Warp Message from Chain A to Chain B
-	ginkgo.It("Verify Message from A to B", ginkgo.Label("Warp", "VerifyMessage"), func() {
-		ctx := context.Background()
+func (w *warpTest) executeHardHatTest() {
+	require := require.New(ginkgo.GinkgoT())
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	client := w.sendingSubnetClients[0]
+	log.Info("Subscribing to new heads")
+	newHeads := make(chan *types.Header, 10)
+	sub, err := client.SubscribeNewHead(ctx, newHeads)
+	require.NoError(err)
+	defer sub.Unsubscribe()
+
+	chainID, err := client.ChainID(ctx)
+	require.NoError(err)
+
+	rpcURI := toRPCURI(w.sendingSubnetURIs[0], w.sendingSubnet.BlockchainID.String())
+
+	os.Setenv("SENDER_ADDRESS", crypto.PubkeyToAddress(fundedKey.PublicKey).Hex())
+	os.Setenv("SOURCE_CHAIN_ID", "0x"+w.sendingSubnet.BlockchainID.Hex())
+	os.Setenv("PAYLOAD", "0x"+common.Bytes2Hex(testPayload))
+	os.Setenv("EXPECTED_UNSIGNED_MESSAGE", "0x"+hex.EncodeToString(w.addressedCallUnsignedMessage.Bytes()))
+	os.Setenv("CHAIN_ID", fmt.Sprintf("%d", chainID.Uint64()))
+
+	cmdPath := "./contracts"
+	// test path is relative to the cmd path
+	testPath := "./test/warp.ts"
+	utils.RunHardhatTestsCustomURI(ctx, rpcURI, cmdPath, testPath)
+}
 
-		log.Info("Subscribing to new heads")
-		newHeads := make(chan *types.Header, 10)
-		sub, err := chainBWSClient.SubscribeNewHead(ctx, newHeads)
-		gomega.Expect(err).Should(gomega.BeNil())
-		defer sub.Unsubscribe()
+func (w *warpTest) warpLoad() {
```

Comment:
Seems this load test waits for all send warp messages to be issued before issuing receiving txs and does not measure throughput or end to end warp message latency. Did we not want to measure these?
For example, in https://github.com/ava-labs/subnet-evm/issues/449 you suggest that txs be sent to a message as they are accepted on the sending chain.

Reply:
I expected you could do something like this: https://github.com/ava-labs/subnet-evm/pull/1010/commits/51b515ed50b9db3edeef193aabb441ddfdf89c3e but this ends up failing when performed concurrently since some nodes have not accepted the block yet at that point.

Adding a TODO for this, but don't think that needs to be addressed now.